index.js

```diff
@@ -8,41 +7,31 @@ class CancelError extends Error {
 	}
 }
 
-const generate = willResolve => function (ms, value) {
-	ms = ms || 0;
-	const useValue = (arguments.length > 1);
-	let result = value;
-
-	const delaying = defer();
-	const promise = delaying.promise;
-
-	let timeout = setTimeout(() => {
-		const settle = willResolve ? delaying.resolve : delaying.reject;
-		settle(result);
-	}, ms);
-
-	const thunk = thunkResult => {
-		if (!useValue) {
-			result = thunkResult;
-		}
-		return promise;
-	};
-
-	thunk.then = promise.then.bind(promise);
-	thunk.catch = promise.catch.bind(promise);
-	thunk._actualPromise = promise;
-
-	thunk.cancel = () => {
-		if (timeout) {
-			clearTimeout(timeout);
-			timeout = null;
-			delaying.reject(new CancelError('Delay canceled'));
+const createDelay = willResolve => (ms, value) => {
+	let timeoutId = null;
+	let internalReject = null;
```

Comment:
No need to assign `null`. Just do:

```js
let timeoutId;
let internalReject;
```

Reply:
Hmm, not sure why I did that.  I thought I was staying consistent with your code but your code never initializes to null.